source/jormungandr/tests/ptref_tests.py

```diff
@@ -778,6 +778,121 @@ def test_pt_objects_with_disable_disruption(self):
         )
         assert len(response['disruptions']) == 1
 
+    def test_networks_with_external_code(self):
+        """test networks with external_code parameter"""
+        response = self.query_region("networks")
+        networks = get_not_null(response, 'networks')
+        assert len(networks) == 4
+        assert networks[0]['id'] == 'network:A'
+        assert networks[1]['id'] == 'network:B'
+        assert networks[2]['id'] == 'network:CaRoule'
+        assert networks[3]['id'] == 'network:freq'
+
+        response = self.query_region("networks?external_code=A")
+        networks = get_not_null(response, 'networks')
+        assert len(networks) == 1
+        assert networks[0]['id'] == 'network:A'
+
+        response = self.query_region("networks?external_code=B")
+        networks = get_not_null(response, 'networks')
+        assert len(networks) == 1
+        assert networks[0]['id'] == 'network:B'
+
+        response = self.query_region("networks?external_code=wrong_code")
+        networks = get_not_null(response, 'networks')
+        assert len(networks) == 4
+
+        # without coverage
+        response, code = self.query_no_assert("v1/networks?external_code=A")
+        assert code == 200
+        networks = get_not_null(response, 'networks')
+        assert len(networks) == 1
+        assert networks[0]['id'] == 'network:A'
+
+        response, code = self.query_no_assert("v1/networks?external_code=wrong_code")
+        assert code == 404
+        message = get_not_null(response, 'message')
+        assert 'Unable to find an object for the uri wrong_code' in message
+
+        # Without coverage, networks have to work with external_code
+        response, code = self.query_no_assert("v1/networks")
+        assert code == 400
+        message = get_not_null(response, 'message')
+        assert 'parameter \"external_code\" invalid: Missing required parameter' in message
+
+    def test_lines_with_external_code(self):
+        """test lines with external_code parameter"""
+        response = self.query_region("lines")
+        lines = get_not_null(response, 'lines')
+        assert len(lines) == 4
+        assert lines[0]['id'] == 'line:A'
+        assert lines[1]['id'] == 'line:B'
+        assert lines[2]['id'] == 'line:Ça roule'
+        assert lines[3]['id'] == 'line:freq'
+
+        response = self.query_region("lines?external_code=A")
+        lines = get_not_null(response, 'lines')
+        assert len(lines) == 1
+        assert lines[0]['id'] == 'line:A'
+
+        response = self.query_region("lines?external_code=wrong_code")
+        lines = get_not_null(response, 'lines')
+        assert len(lines) == 4
```

Comment:
I don't understand that (I didn't pay attention: same for network and stop_point).
The parameter is OK and the value is not, so we should have nothing, no?
From a user point-of-view the comparison with the response without coverage doesn't make sense (and we probably should get no object here).

Reply:
I think you've right. it should be the same behavior. Fix it fix it